src/System.IO.Pipes/src/System/IO/Pipes/AnonymousPipeServerStream.Unix.cs

```diff
@@ -18,9 +17,56 @@ public sealed partial class AnonymousPipeServerStream : PipeStream
         private void Create(PipeDirection direction, HandleInheritability inheritability, int bufferSize)
         {
             Debug.Assert(direction != PipeDirection.InOut, "Anonymous pipe direction shouldn't be InOut");
-            Debug.Assert(bufferSize >= 0, "bufferSize is negative");
+            // Ignore bufferSize.  It's optional, and the fcntl F_SETPIPE_SZ for changing it is Linux specific.
 
-            throw NotImplemented.ByDesign; // TODO: Implement this
+            // Use pipe or pipe2 to create our anonymous pipe
+            int[] fds = new int[2];
+            unsafe
+            {
+                fixed (int* fdsptr = fds)
+                {
+                    bool created = false;
+                    
+                    // If the caller asked for the handle to be non-inheritable, try to use pipe2 to create the pipe.
+                    // Depending on the OS, it may not exist.
+                    if ((inheritability & HandleInheritability.Inheritable) == 0)
+                    {
+                        try
+                        {
+                            while (Interop.CheckIo(Interop.libc.pipe2(fdsptr, (int)Interop.libc.OpenFlags.O_CLOEXEC))) ;
+                            created = true;
+                        }
+                        catch (MissingMethodException) { } // pipe2 may not exist
```

Comment:
I agree it's questionable, but I went this way because having an entirely separate binary just to be able to handle one enum value option felt like overkill.  We do similar "light-up" in other places in CoreFX, such as in the metadata reader, and this seemed like an equivalent case.